platform/engineering/request-for-comment/2022-01-13-switch-to-vaos-v2-appointments.md

```diff
@@ -0,0 +1,137 @@
+# RFC: Switch VA Mobile app to VAOS V2 appointments
+
+- Date: 2022-01-13
+- Related Issue: https://github.com/department-of-veterans-affairs/va-mobile-app/issues/2227
+
+## Background
+The VA mobile app fetches veterans' VA appointment histories from the VAOS V0 appointments endpoint and their Community Care (CC) clinic appointment history from the community care appointments endpoint. VAOS recently developed a V2 appointments endpoint that includes both VA and CC appointments. We would like to switch to using the new endpoint before the old one is deprecated.
+
+## Motivation
+The VAOS V0 appointments endpoint is being deprecated.
+
+## Design
+
+The current mobile appointment data comes from two sources: the VAOS V0 appointments endpoint and Community Cares endpoints. The appointments controller uses the appointments proxy to gather appointments from both sources, normalize the data via adapters, and interleaves the results.
+
+The VA appointments endpoint only includes the facility ID without its name and other relevant information, so the proxy also makes a request to gather facility data, which is used to complete the appointments data. The Community Care appointments endpoint includes facility data.
+
+The V2 appointments endpoint includes both VA and CC appointments, so we should no longer need to fetch CC appointments separately. The current mock data for CC appointments is not complete and does not include embedded clinic information like the current endpoint does. That information may later be added, but if it isn't, we may need to make a secondary request for that information via the VAOS clinic endpoint.
+
+This effort will require:
+- small changes to the proxy
+- rewriting the VA appointments adapter
+- removing or rewriting the CC appointments adapter
+- changes to the mobile appointments service to use the new url
+- possibly adding a new service for fetching clinic data
+- substantial effort to ensure that the data normalization process is correct and matches the current data sent to the mobile app. See the mapping sections below for an initial attempt to map incoming data to the existing mobile appointment payload.
+
+### VA Appointments Mapping:
```

Comment:
Not sure why but the mapping sections are rendering as one big blob of text with no line breaks. The content here is good though, nice to have this level of detail about each field 👍

Reply:
Huh. Strange. I don't see it that way. I'm using firefox. Not sure why it would matter though.